### PR TITLE
Implementation of new state-of-the-art LLM optimizer: The AdEMAMix Optimizer

### PR DESCRIPTION
**Pull request by ovuruska**
Implement issue #135609

Here is the [link](https://arxiv.org/abs//2409.03137) for the original paper.

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar

### CODE REVIEW

File: torch/optim/__init__.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 from torch.optim.adam import Adam as Adam
 from torch.optim.adamax import Adamax as Adamax
 from torch.optim.adamw import AdamW as AdamW
+from torch.optim.ademamix import AdEMAMix as AdEMAMix
```
Comment (PhilipMay) [inline torch/optim/__init__.py:16]: I think this could be rewritten to just:

`from torch.optim.ademamix import AdEMAMix`

**Comment by ovuruska** [comment]
Hi @vadimkantorov , any updates on this?

**Comment by vadimkantorov** [comment]
@ovuruska I'm not a maintainer, can't advise on this...

**Comment by ovuruska** [comment]
@vadimkantorov No problem, thanks for letting me know.

**Comment by xcgoner** [comment]
@ovuruska I've read your code. There seems to be a lot of typos which make your implementation very different from the original paper. Please double check.
For example, the calculation of beta3_t (mostly exponent) doesn't seem right.

**Comment by ovuruska** [comment]
Hi @xcgoner , thanks for the feedback.

**Pull request closed without merging**